.github/workflows/test.yml

```diff
@@ -0,0 +1,52 @@
+name: Run the test suite
+on:
+  push:
+    branches:
+      - master
+      - develop
+  pull_request:
+
+jobs:
+  test-on-linux:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["2.7", "3.5", "3.6", "3.7", "3.8", "3.9", "3.10"]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install the project and its dependencies
+        run: pip install -r requirements-dev.txt
+      - name: Test with pytest
+        run: pytest --cov=. --cov-report xml --doctest-modules --cov-report term
+      - name: Test Tornado application
+        run: python -m unittest discover -v -p '*tornado*' tests
+
+  test-on-mac-and-windows:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, windows-latest]
+        python-version: ["2.7", "3.5", "3.6", "3.7", "3.8", "3.9", "3.10"]
+        exclude:
+          - os: windows-latest
+            python-version: "3.10"  # won't compile psycopg2
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install the project and its dependencies
+        run: pip install -r requirements-dev.txt
+      - name: Test with pytest
+        run: pytest --cov=. --cov-report xml --doctest-modules --cov-report term
+      - name: Test flask application
+        run: python -m unittest discover -v -p '*flask*' tests
+      - name: Test Tornado application
+        run: python -m unittest discover -v -p '*tornado*' tests
```

Comment:
I'd suggest to call tox here instead of duplicating the code from [tox.ini](https://github.com/ateliedocodigo/py-healthcheck/blob/aae17b08571ca7320b0c01297c58c3eab92316f0/tox.ini#L30):
* First install tox together with `requirements-dev.txt`
* Then call `tox -e flask` and `tox -e tornado`

That way devs can easily reproduce errors that might occur in the Github Action, simply by calling tox. Also the tox settings and the Github Action Checks would be inherently consistent.